server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteRoutingTableServiceIT.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    private BlobPath getIndexRoutingPath(BlobPath indexRoutingPath, String indexUUID) {
+        RemoteStoreEnums.PathHashAlgorithm pathHashAlgo = RemoteStoreEnums.PathHashAlgorithm.FNV_1A_BASE64;
+        return pathType.path(
+            RemoteStorePathStrategy.PathInput.builder().basePath(indexRoutingPath).indexUUID(indexUUID).build(),
+            pathHashAlgo
+        );
+    }
+
+    private void verifyUpdatesInManifestFile(RemoteManifestManager remoteManifestManager) {
+        Optional<ClusterMetadataManifest> latestManifest = remoteManifestManager.getLatestClusterMetadataManifest(
+            getClusterState().getClusterName().value(),
+            getClusterState().getMetadata().clusterUUID()
+        );
+        assertTrue(latestManifest.isPresent());
+        ClusterMetadataManifest manifest = latestManifest.get();
+        assertTrue(manifest.getDiffManifest().getIndicesRoutingUpdated().contains(INDEX_NAME));
+        assertTrue(manifest.getDiffManifest().getIndicesDeleted().isEmpty());
+        assertFalse(manifest.getIndicesRouting().isEmpty());
+        assertEquals(1, manifest.getIndicesRouting().size());
+        assertTrue(manifest.getIndicesRouting().get(0).getUploadedFilename().contains(indexRoutingPath.buildAsString()));
+    }
+
+    private List<RoutingTable> getRoutingTableFromAllNodes() throws ExecutionException, InterruptedException {
+        String[] allNodes = internalCluster().getNodeNames();
+        List<RoutingTable> routingTables = new ArrayList<>();
+        for (String node : allNodes) {
+            RoutingTable routingTable = internalCluster().client(node)
+                .admin()
+                .cluster()
+                .state(new ClusterStateRequest().local(true))
+                .get()
+                .getState()
+                .routingTable();
+            routingTables.add(routingTable);
+        }
+        return routingTables;
+    }
+
+    private boolean areRoutingTablesSame(List<RoutingTable> routingTables) {
+        if (routingTables == null || routingTables.isEmpty()) {
+            return false;
+        }
+
+        RoutingTable firstRoutingTable = routingTables.get(0);
+        for (RoutingTable routingTable : routingTables) {
+            if (!firstRoutingTable.equals(routingTable)) {
+                logger.info("Responses are not the same: {} {}", firstRoutingTable, routingTable);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void updateIndexSettings(String indexName, String settingKey, int settingValue) {
+        client().admin()
+            .indices()
+            .prepareUpdateSettings(indexName)
+            .setSettings(Settings.builder().put(settingKey, settingValue))
+            .execute()
+            .actionGet();
+    }
+
+    private void deleteIndexAndVerify(RemoteManifestManager remoteManifestManager, NodesStatsResponse nodesStatsResponse) {
```

Comment:
Are we using nodesStatesResponse in this method?

Reply:
No, removed it.